### DISCUSSION TITLE
Include forwarded headers by default

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#1909] [Ideas]
I'm in the process of getting the deployment of an aspire application working and one thing I hit was that authentication wasn't working with OIDC, the redirect url it was sending was http rather than https. I've used 'azd infra synth' to generate bicep and yaml files for the project.

There's an environment variable for .NET apps in container environments: ASPNETCORE_FORWARDEDHEADERS_ENABLED which needs to be set to true for external web apps in container environments for things like OIDC to work ([link](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-8.0#forward-the-scheme-for-linux-and-non-iis-reverse-proxies)).

Maybe this would be a sensible thing to include by default? I can do it like this:

```
var apiservice = builder.AddProject<Projects.ecoDriverAnalytics_Api>("ecodriver-analytics-api")
	.WithEnvironment("ASPNETCORE_FORWARDEDHEADERS_ENABLED", "true")
```

But maybe it should do that under the skin automatically?

**Comment by davidfowl** [comment]
See https://github.com/dotnet/aspire/issues/290
